instill/model/v1alpha/model.proto

```diff
@@ -0,0 +1,403 @@
+    // Serving status: NOT SERVING
+    SERVING_STATUS_NOT_SERVING = 2;
+  }
+
+  // Status is the value of ServingStatus enum type
+  ServingStatus status = 1;
+}
+
+// ModelVersion represents the version of a model
+message ModelVersion {
+
+  // Status enumerates the status of a model version
+  enum Status {
+    // Status: UNSPECIFIED
+    STATUS_UNSPECIFIED = 0;
+    // Status: OFFLINE
+    STATUS_OFFLINE = 1;
+    // Status: ONLINE
+    STATUS_ONLINE = 2;
+    // Status: ERROR
+    STATUS_ERROR = 3;
+  }
+
+  // Model version
+  uint64 version = 1;
+  // Model version corresponding model ID
+  uint64 model_id = 2;
+  // Model version description
+  string description = 3;
+  // Model version creation time
+  google.protobuf.Timestamp created_at = 4;
+  // Model version update time
+  google.protobuf.Timestamp updated_at = 5;
+  // Model version status
+  Status status = 6;
+}
+
+// Model represents the content of a model
+message Model {
+
+  // Task enumerates the task type of a model
+  enum Task {
+    // Task: UNSPECIFIED
+    TASK_UNSPECIFIED = 0;
+    // Task: CLASSIFICATION
+    TASK_CLASSIFICATION = 1;
+    // Task: DETECTION
+    TASK_DETECTION = 2;
+  }
+
+  // Model ID
+  uint64 id = 1 [ (google.api.field_behavior) = OPTIONAL ];
+  // Model name
+  string name = 2;
+  // Model full name (i.e., [user_name/org_name]/model_name)
+  string full_name = 3 [ (google.api.field_behavior) = OUTPUT_ONLY ];
+  // Model task
+  Task task = 4;
+  // Model versions
+  repeated ModelVersion model_versions = 5;
+}
+
+// ModelInitData represents the required data for initialising a model instance
+message ModelInitData {
+  // Model name
+  string name = 1 [ (google.api.field_behavior) = REQUIRED ];
+  // Model content in byte
+  bytes byte = 2 [ (google.api.field_behavior) = REQUIRED ];
+  // Model description
+  string description = 3 [ (google.api.field_behavior) = OPTIONAL ];
+  // Model task type
+  Model.Task task = 4 [ (google.api.field_behavior) = OPTIONAL ];
+}
+
+// CreateModelRequest represents a request to create a model
+message CreateModelRequest {
+  // Model initialising data
+  ModelInitData model_init_data = 1 [ (google.api.field_behavior) = REQUIRED ];
+}
+
+// CreateModelResponse represents a response for a model instance
+message CreateModelResponse {
+  // A model instance
+  Model model = 1 [ (google.api.field_behavior) = REQUIRED ];
+}
+
+// CreateModelBinaryFileUploadRequest represents a request to create a model
+message CreateModelBinaryFileUploadRequest {
+  // Model initialising data
+  ModelInitData model_init_data = 1 [ (google.api.field_behavior) = REQUIRED ];
+}
+
+// CreateModelBinaryFileUploadResponse represents a response for a model
+// instance
+message CreateModelBinaryFileUploadResponse {
+  // A model instance
+  Model model = 1 [ (google.api.field_behavior) = REQUIRED ];
+}
+
+// CreateModelRequest represents a request to update a model
+message UpdateModelVersionRequest {
```

Comment:
@pinglin @xiaofei-du for updating pipeline, I see that we use Pipeline object in request. So for more consistency, it should be same in update model version (i.e use ModelVersion in the request instead of separated fields such as description, status). What do you think?
For this approach,@bochengyang @xiaofei-du also discussed before, there are some fields in pipeline/modelversion that could not be updated.

Reply:
That's true and modified accordingly. :)